cmake/modules/FindFUSE.cmake

```diff
@@ -28,14 +28,28 @@ find_library(FUSE_LIBRARIES
   NAMES ${fuse_names}
   PATHS /usr/local/lib64 /usr/local/lib)
 
-foreach(ver "MAJOR" "MINOR")
-  file(STRINGS "${FUSE_INCLUDE_DIR}/fuse_common.h" fuse_ver_${ver}_line
-    REGEX "^#define[\t ]+FUSE_${ver}_VERSION[\t ]+[0-9]+$")
-  string(REGEX REPLACE ".*#define[\t ]+FUSE_${ver}_VERSION[\t ]+([0-9]+)$"
-    "\\1" FUSE_VERSION_${ver} "${fuse_ver_${ver}_line}")
-endforeach()
+set(ENV{PKG_CONFIG_PATH} $ENV{PKG_CONFIG_PATH}:/usr/local/lib64/pkgconfig)
```

Comment:
i am curious why you want to add `/usr/local/lib64/pkgconfig` to `PKG_CONFIG_PATH`?

Reply:
For this because I need to debug and test all the possible versions in my local node and installed from source.

Reply:
if we can set this env variable when calling cmake, probably we can dispense with this setting? as i think we should offer a machinery which is general enough instead of hardwiring user settings in the code.

Reply:
How about adding an option for the CMakeList.txt ?
such as in CMakeList.txt add:

```
set(ENV{PKG_CONFIG_PATH} $ENV{PKG_CONFIG_PATH}:${devpath}/lib64/pkgconfig)
```
And for find_path/find_library:
```
find_path(
...
PATHS ${devpath}
)

find_library(
...
PATHS ${devpath}
)
```


Then `./do_cmake.sh -Ddevpath=/usr/local/`

Reply:
i'd prefer just use
```
PKG_CONFIG_PATH=/my/install/path/of/libraries/pkgconfig ./do_cmake.sh
```

Reply:
But this still won't work to find the library and header files in `/my/install/path/of/libraries/`. It will only work for finding the pkgconfig files.

Such as the for the find_library, currently it is specifying the local PATHS as default:

```
 27 find_library(FUSE_LIBRARIES                                                                                                            
 28   NAMES ${fuse_names}
 29   PATHS /usr/local/lib64 /usr/local/lib)
```

The `/usr/local/` should be common as default for developers to install the sources.

Reply:
firstlt, this change is orthogonal to the problem you want to address here. the proposal you put above to enable FindFUSE.cmake to find library in specified directory, so i recommend move this change into its own commit and have something like
```cmake
find_package(PkgConfig QUIET)
pkg_search_module(PC_FUSE QUIET ${fuse_names})
find_path(FUSE_INCLUDE_DIR
   NAMES fuse_common.h fuse_lowlevel.h fuse.h
   HINTS
     ${PC_FUSE_INCLUDE_DIRS}
   PATH_SUFFIXES ${fuse_suffixes})
find_library(FUSE_LIBRARIES
  NAMES ${fuse_names}
  HINTS
     ${PC_FUSE_LIBRARY_DIRS}
  PATHS /usr/local/lib64 /usr/local/lib)
```

Reply:
Okay, works and fixed it.

Reply:
please add this in a separated commit.

Reply:
Done, please check.
Thanks.